lib/user_space/uuuid.c

```diff
@@ -73,24 +75,30 @@ int m0_node_uuid_string_get(char buf[M0_UUID_STRLEN + 1])
 {
 	int fd;
 	int rc = 0;
+	char mid[MACHINE_ID_LEN + 1];
+
+	M0_BASSERT(MACHINE_ID_LEN == M0_UUID_STRLEN - 4);
 
 	if (use_default_node_uuid) {
 		strncpy(buf, default_node_uuid, M0_UUID_STRLEN + 1);
 		buf[M0_UUID_STRLEN] = '\0';
 	} else {
-		fd = open(kmod_uuid_file, O_RDONLY);
+		fd = open(uuid_file, O_RDONLY);
 		if (fd < 0) {
 			return M0_ERR_INFO(-EINVAL,
 			                   "open(\"%s\", O_RDONLY)=%d "
-			                   "errno=%d: is m0tr.ko loaded?",
-					   kmod_uuid_file, fd, errno);
+			                   "errno=%d: ",
+					   uuid_file, fd, errno);
 		}
-		if (read(fd, buf, M0_UUID_STRLEN) == M0_UUID_STRLEN) {
+		if (read(fd, mid, MACHINE_ID_LEN) == MACHINE_ID_LEN) {
 			rc = 0;
+			snprintf(buf, 37, "%.8s-%.4s-%.4s-%.4s-%.12s", mid,
```

Comment:
Please replace the "37" with "M0_UUID_STRLEN + 1", that is the size of the buf.

Reply:
incorporated